api/utils/error_parsing.ts

```diff
@@ -0,0 +1,100 @@
+interface IErrorLocation {
+  line?: number;
+  column?: number;
+}
+
+interface IAzureEvalError {
+  originalError?: {
+    info?: {
+      message?: string;
+    };
+  };
+  errorLocation?: IErrorLocation;
```

Comment:
this seems a bit weird to me. we mutate the original `Error`, instead of just returning a new `EvaluationError` instance.

Reply:
Updated all to create a new instance.